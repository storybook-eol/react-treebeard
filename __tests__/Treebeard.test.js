import React from 'react';
import {shallow} from 'enzyme';

import Treebeard from '../src/components';
import data from './mocks/data';

const renderComponent = (props = {}) => {
    const wrapper = shallow(
        <Treebeard data={props.data || data} {...props}/>
    );
    wrapper.treeNode = () => wrapper.find('TreeNode');
    return wrapper;
};

describe('<Treebeard/>', () => {
    it('should match default snapshot', () => {
        const wrapper = renderComponent();
        expect(wrapper).toMatchSnapshot();
    });
    it('should handle custom makeKeyProp', () => {
        const wrapper = renderComponent({ makeKeyProp: (node) => node.sha });
        const firstNodeKey = wrapper.find('TreeNode').first().key();
        expect(firstNodeKey).toBe('9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08');
    });
});
