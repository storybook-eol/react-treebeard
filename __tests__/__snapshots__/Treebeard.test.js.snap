// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Treebeard/> should match default snapshot 1`] = `
<Styled(ul)
  style={
    Object {
      "backgroundColor": "#21252B",
      "color": "#9DA5AB",
      "fontFamily": "lucida grande ,tahoma,verdana,arial,sans-serif",
      "fontSize": "14px",
      "listStyle": "none",
      "margin": 0,
      "padding": 0,
    }
  }
>
  <TreeNode
    animations={
      Object {
        "drawer": [Function],
        "toggle": [Function],
      }
    }
    customStyles={Object {}}
    decorators={
      Object {
        "Container": [Function],
        "Header": [Function],
        "Loading": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": [Function],
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            [Function],
          ],
          "defaultProps": undefined,
          "displayName": "Styled(Component)",
          "propTypes": Object {
            "style": [Function],
          },
          "render": [Function],
          "withComponent": [Function],
        },
        "Toggle": [Function],
      }
    }
    key="1"
    makeKeyProp={[Function]}
    node={
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "name": "app.js",
                "sha": "828270ffd0e8ee6f32eb2a83bc3da31f3a9ac73f4833de481f65337c0802a9aa",
              },
              Object {
                "name": "data.js",
                "sha": "3ae815b93875786c05cba19df444662a789640e83ab65f4b84e9c16474cb1ed0",
              },
              Object {
                "name": "index.html",
                "sha": "f82ec4ae606600a57396ca47e21e0f013bded8523dfaffee040eefa4ea815962",
              },
              Object {
                "name": "styles.js",
                "sha": "3ba988e33afe3c6e453fc86bf26813ad90e5fe2dd7f55527a8c1408f861bfeac",
              },
              Object {
                "name": "webpack.config.js",
                "sha": "017cbc93d594c0f410b0c4c86c6b160a421674ddf72ab10201a16646f02fcb9c",
              },
            ],
            "name": "example",
            "sha": "f05bcd2dffa3e047ce58c0f1a1877dcbe4d868cf0eac58bc2f7f9ab805289b0b",
          },
          Object {
            "children": Array [],
            "loading": true,
            "name": "node_modules",
            "sha": "bd9d2e8d246bc323081327ef12d3d40e0fc09cc7d36ea05bc286ea74e47c3515",
          },
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "name": "decorators.js",
                    "sha": "cc37c52929a8f16cfbf7b1881119f963e5a332507fac7d7f4fa9a271ad58534e",
                  },
                  Object {
                    "name": "treebeard.js",
                    "sha": "177c1b4a2b85d1c2a15ed481dbcefd9a43a901d6efa05c41dbb7e1859d5547e2",
                  },
                ],
                "name": "components",
              },
              Object {
                "name": "index.js",
                "sha": "cc02786ca507e473781c0c08a2738e921eaa49c061515598d0ef1dff3054ea7b",
              },
            ],
            "name": "src",
            "sha": "da5bd1bf868d4b2a68ce84c5e13545523ddc58232fe59e74a3d8ccb9be9e6db6",
          },
          Object {
            "children": Array [
              Object {
                "name": "animations.js",
                "sha": "1fde2cd5a18a6bd17660167f0fe5296a2ea1275da38961632b5630abf9c7a85d",
              },
              Object {
                "name": "default.js",
                "sha": "a59dc347bb2eb8f5a51f3b7f65dd63580829bcdfee89d5331f9ff063438b6dbd",
              },
            ],
            "name": "themes",
            "sha": "8c382185c33911dbc2e2a76094b73a02644635ed9de1e9995a8b06f96bdbae32",
          },
          Object {
            "name": "gulpfile.js",
            "sha": "f68fe2b19685fb03204cfb3a0c7e58c7698fa9abc595185e39028fbc6e0531e1",
          },
          Object {
            "name": "index.js",
            "sha": "08f982599837827c163a787c6ae0a5cf1b493829493c1f29d707a5026b28ed0e",
          },
          Object {
            "name": "package.json",
            "sha": "c31e4a244c319e43a03cd3fbfabd0c6d976567c5b79e8b06a218b2aae3637839",
          },
        ],
        "id": 1,
        "name": "react-treebeard",
        "sha": "9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08",
        "toggled": true,
      }
    }
    style={
      Object {
        "activeLink": Object {
          "background": "#31363F",
        },
        "base": Object {
          "position": "relative",
        },
        "header": Object {
          "base": Object {
            "color": "#9DA5AB",
            "display": "inline-block",
            "verticalAlign": "top",
          },
          "connector": Object {
            "borderBottom": "solid 2px black",
            "borderLeft": "solid 2px black",
            "height": "12px",
            "left": "-21px",
            "position": "absolute",
            "top": "0px",
            "width": "2px",
          },
          "title": Object {
            "lineHeight": "24px",
            "verticalAlign": "middle",
          },
        },
        "link": Object {
          "cursor": "pointer",
          "display": "block",
          "padding": "0px 5px",
          "position": "relative",
        },
        "loading": Object {
          "color": "#E2C089",
        },
        "subtree": Object {
          "listStyle": "none",
          "paddingLeft": "19px",
        },
        "toggle": Object {
          "arrow": Object {
            "fill": "#9DA5AB",
            "strokeWidth": 0,
          },
          "base": Object {
            "display": "inline-block",
            "height": "24px",
            "marginLeft": "-5px",
            "position": "relative",
            "verticalAlign": "top",
            "width": "24px",
          },
          "height": 14,
          "width": 14,
          "wrapper": Object {
            "height": "14px",
            "left": "50%",
            "margin": "-7px 0 0 -7px",
            "position": "absolute",
            "top": "50%",
          },
        },
      }
    }
  />
</Styled(ul)>
`;
